clarity/contracts/arkadiko-stake-registry-v1-1.clar

```diff
@@ -88,26 +107,10 @@
 ;; )
 
 ;; Claim pool rewards
-(define-public (claim-pending-rewards (pool-trait <stake-pool-trait>))
+(define-public (claim-pending-rewards (registry-trait <stake-registry-trait>) (pool-trait <stake-pool-trait>))
   (begin
-    (contract-call? pool-trait claim-pending-rewards tx-sender)
-  )
-)
-
-;; Mint rewards for staker - for active pools only
-(define-public (mint-rewards-for-staker (amount uint) (staker principal))
-  (begin
-    (let (
-      (pool-data (get-pool-data contract-caller))
-    )
-      ;; Only active pools can mint rewards
-      (asserts! (is-eq (get active pool-data) true) ERR-POOL-INACTIVE)
-
-      ;; Mint DIKO rewards for staker
-      (try! (contract-call? .arkadiko-dao mint-token .arkadiko-token amount staker))
-      
-      (ok amount)
-    )
+    (asserts! (is-eq (as-contract tx-sender) (contract-of registry-trait)) ERR-WRONG-REGISTRY)
```

Comment:
Q: all of these methods that have this `asserts!` cannot be called externally from a random addr, right?

Q2: on line 93 we still have a completely commented out method, do we still need that?

Reply:
A: correct, the parameter must be the current registry. Will add some tests.
A2: it's not really needed as you can call the method on the pool directly. But it's nice to have all user interactions done via the registry, so will make sure to add the method and tests.

Reply:
Both addressed in last commit.